File: false_position_method.nim
import math
const acc = 1.0e-7

proc f(x: float): float =
    return sin(x) - x + 1

proc bisection(a1, b1: float) =
    var
        a = a1
        b = b1
    if f(a1)*f(b1) >= 0:
        echo "You did't assumed right the root is between a and b\n"
        return
    var c: float
    while (b-a) >= acc:
        c = (a*f(b)-b*f(a)) / (f(b)-f(a))
        if f(c) == 0:
            break
        elif f(a)*f(c) < 0:
            b = c
        else:
            a = c
    echo "the value of root is ", c;

bisection(0.0, 4.0)

File: false_position_method.cpp
#include <iostream>
#include <cmath>
#include <iomanip>
#define EPSILON 1.0e-7
using namespace std;

double f(double x)
{
    return sin(x) - x + 1;
}

void regula_falsi(double a, double b)
{
    if (f(a) * f(b) >= 0)
    {
        cout << "You did't assumed right the root is between a and b\n";
        return;
    }
    double c;
    while ((b - a) >= EPSILON)
    {
        c = (a * f(b) - b * f(a)) / (f(b) - f(a));
        if (f(c) == 0)
        {
            break;
        }
        else if (f(c) * f(a) < 0)
        {
            b = c;
        }
        else
        {
            a = c;
        }
    }
    cout << "the value of root is " << setprecision(20) << c;
}

int main()
{
    regula_falsi(0, 4);
    return 0;
}